### PR TITLE
Fix sesv2 from_email_address missing sender name

### PR DESCRIPTION
**Pull request by sgomez17**
*Issue #97*

*Description of changes:*
Fix the issue reported in #97 for SESV2.

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/aws/rails/sesv2_mailer.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def initialize(options = {})
       def deliver!(message)
         params = {
           content: { raw: { data: message.to_s } },
-          from_email_address: message.smtp_envelope_from, # defaults to From header
+          from_email_address: from_email_address(message),
           # defaults to destinations (To,Cc,Bcc) - stick all on bcc to be delivered anyway
           destination: { bcc_addresses: message.smtp_envelope_to }
```
Comment (mullermp) [inline lib/aws/rails/sesv2_mailer.rb:32]: Do we need a similar fix to smtp_envelope_to somehow?

**Comment by mullermp** [comment]
I'm doing some testing today. I'm not sure about this approach currently. The correct sender name appears when I set the From header with a name and email address AND omit `smtp_envelope_from`. We may actually want to check `mail.from_addrs.first != mail.smtp_envelope_from`, and when true, use `mail.smtp_envelope_from` as ses v2 `from_email_address` param, or omit it otherwise.

**Comment by mullermp** [comment]
I think I'm leaning towards a solution that uses `instance_variable_get` on smtp_envelope_to/from values and omitting / adapting certain SES v2 params.

**Comment by mullermp** [comment]
Is the SESv2 mailer working with Bcc? I thought we solved that problem already, but in all of my testing, and even just using what's coded now, it's not working... SES v2 is so finicky :/

**Comment by mullermp** [comment]
@sgomez17 I pushed a change with the `instance_variable_get` approach. I think the defaults for `smtp_envelope_from/to` methods are not ideal for SES v2's API design, but work well for SES v1. I think in both APIs, SES will do comparisons and defaults to find the correct names and addresses. We can check if the instance variable was set (they were explicitly set) and if they were, use them as `from_email_address` and `destinations[:to_addresses]`. Otherwise, we omit `from_email_address` (SES uses From header by default, including name) and we use To header value. I think this is correct because SES v2 documents all of these members as optional - because raw message is supposed to define all of these ourselves.

Please try this mailer out and see if it works for you! I did some testing on my side and it seems to work as expected - I see names for To, From, and Cc, and also sends to Bcc.

**Comment by sgomez17** [comment]
hi @mullermp, I have tested it and yes it and that latest version works well for our use-case, so it's a thumbs up from me to merge :+1:

**Pull request merged**